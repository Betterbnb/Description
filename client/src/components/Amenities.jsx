import React from 'react';
import styled from 'styled-components';

const StyledAmenities = styled.div`
  font-family: Circular, Helvetica Neue, Helvetica, Arial, sans-serif;
  color: #484848;
  margin: 20px 20px 10px 10px;
  font-size: 15px;
  font-weight: 320;
  line-height: 1.4;
  letter-spacing: 0.5px;
`;

const title = {
  fontSize: '16px',
  fontWeight: '500',
  marginBottom: '5px'
};

const Amenities = () => {
  return(
    <StyledAmenities style={title}>Amenities</StyledAmenities>
  )
}

export default Amenities;