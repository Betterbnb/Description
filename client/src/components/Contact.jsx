import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledContact = styled.button`
  font-family: Circular, Helvetica Neue, Helvetica, Arial, sans-serif;
  color: #00abb2;
  border: none;
  outline: none;
  padding: 0;
  margin: 20px 20px 10px 10px;
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 0.5px;
  align-self: flex-start;
`;

const Backdrop = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.8);
  z-index: 1;
`;

const EmailContact = styled.div`
  font-family: Circular, Helvetica Neue, Helvetica, Arial, sans-serif;
  box-shadow: 0 0 10px #e6e6e6;
  background-color: #fff;
  color: #484848;
  padding: 25px;
  height: auto;
  max-height: max-content;
  font-weight: 300;
  line-height: 1.5;
`;

const CloseButton = styled.input`
  box-shadow: 0 0 4px #b7b7b7;
  border-radius: 50%;
  height: 24px;
  width: 24px;
  padding: 5px;
  margin-bottom: 20px;
  outline: none;
  vertical-align: middle;
`;

const imageUrls = {
  close: 'https://s3-us-west-1.amazonaws.com/betterbnb-description/closeButton.png'
};

class Contact extends React.Component {
  constructor(props) {
    super(props);
    this.state = { showModal: false };
  }

  toggleModal() {
    this.setState((prevState) => ({ showModal: !prevState.showModal }));
  }

  render() {
    const { contact } = this.props.owner;
    if (!this.state.showModal) {
      return <StyledContact 
        link={contact} 
        onClick={() => this.toggleModal()}>Contact host</StyledContact>
    } else {
      return(
        <div>
          <StyledContact link={contact}>Contact host</StyledContact>
          <Backdrop>
            <EmailContact>
              <CloseButton type='image' src={imageUrls.close} onClick={() => this.toggleModal()}></CloseButton>
              <div style={{'fontWeight': '400'}}>Email host at:</div>
              <div>{contact}</div>
            </EmailContact>
          </Backdrop>
        </div>
      )
    }
  }
};

Contact.propTypes = {
  owner: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.array, PropTypes.bool])),
  contact: PropTypes.string
};

Contact.defaultProps = {
  owner: {
    name: '',
    image: '',
    contact: '',
    badge: false
  },
  contact: ''
};

export default Contact;