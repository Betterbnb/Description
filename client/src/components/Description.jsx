import React from 'react';
import styled from 'styled-components';

const DescriptionBox = styled.div`
  font-family: Circular, Helvetica Neue, Helvetica, Arial, sans-serif;
  color: #484848;
  margin: 20px 20px 10px 10px;
  font-size: 15px;
  font-weight: 320;
  line-height: 1.4;
  letter-spacing: 0.75px;
`;

const title = {
  fontSize: '16px',
  fontWeight: '400',
  marginBottom: '5px'
};

const link = {
  fontSize: '16px',
  fontWeight: '500',
  color: '#00abb2',
  letterSpacing: '0.5px'
};

// This function is used where paragraph title is stored as camelCase property
// in database. Code may need to be refactored if data schema is altered.

const transformCase = (camelCase) => {
  let output = '';
  camelCase.split('').forEach(letter => {
    if (letter == letter.toUpperCase()) {
      output = output.concat(' ', letter.toLowerCase());
    } else {
      output = output.concat(letter);
    }
  });
  return output.slice(0, 1).toUpperCase() + output.slice(1);
}

class Description extends React.Component {
  constructor(props) {
    super(props);
    this.state = { showText: false };
    this.toggleShowText = this.toggleShowText.bind(this);
  }

  componentDidMount() {}

  toggleShowText() {
    let newState = this.state.showText ? false : true;
    this.setState({ showText: newState });
  }

  render() {
    if (this.state.showText) {
      return(
        <DescriptionBox>
          {
            Object.keys(this.props.description).map((keyName, keyIndex) => {
              return (
                <div key={keyIndex}>
                  <div style={title}>{transformCase(keyName) === 'General' ? '' : transformCase(keyName)}</div>
                  <p>{this.props.description[keyName]}</p>
                </div>
              )
            })
          }
          <a style={link} onClick={() => this.toggleShowText()}>Hide &#9650;</a>
        </DescriptionBox>
      )
    } else {
      return(
        <DescriptionBox>
          <p>{this.props.description.general}</p>
          <a style={link} onClick={() => this.toggleShowText()}>Read more about the space &#9660;</a>
        </DescriptionBox>
      )
    }
  }
}

export default Description;