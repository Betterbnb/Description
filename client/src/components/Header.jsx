import React from 'react';
import CapacityBar from './CapacityBar.jsx';
import styled from 'styled-components';

const StyledHeader = styled.div`
  background: blue,
  color: red
`

const Header = ({ propertyType, name, location, owner, guests, beds, bathrooms }) => {
  return(
    <div>
      <StyledHeader>Header</StyledHeader>
      <div>{propertyType.toUpperCase()}</div>
      <div>{name}</div>
      <div>{location}</div>
      <img src={owner.image}></img>
      <div>{owner.name}</div>
      <CapacityBar guests={guests} beds={beds} bathrooms={bathrooms}/>
    </div>
  )
};

export default Header;