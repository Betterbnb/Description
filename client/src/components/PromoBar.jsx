import React from 'react';

class PromoBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return(
      <h5>PromoBar</h5>
    )
  }
}

export default PromoBar;