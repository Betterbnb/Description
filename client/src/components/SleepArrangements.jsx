import React from 'react';

const SleepArrangements = () => {
  return(
    <h5>SleepArrangements</h5>
  )
}

export default SleepArrangements;