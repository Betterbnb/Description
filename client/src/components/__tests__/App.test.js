import React from 'react';
import { shallow } from 'enzyme';
import App from '../App.jsx';
import Header from '../Header.jsx';
import PromoBar from '../PromoBar.jsx';
import Description from '../Description.jsx';
import Contact from '../Contact.jsx';
import Amenities from '../Amenities.jsx';
import SleepArrangements from '../SleepArrangements';

describe('<App />', () => {

  it('should render correctly with no props', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });

  it('should have subcomponent called Header', () => {
    expect(shallow(<App />).find(Header).length).toEqual(1);
  });

  it('should have subcomponent called PromoBar', () => {
    expect(shallow(<App />).find(PromoBar).length).toEqual(1);
  });

  it('should have subcomponent called Description', () => {
    expect(shallow(<App />).find(Description).length).toEqual(1);
  });

  it('should have subcomponent called Contact', () => {
    expect(shallow(<App />).find(Contact).length).toEqual(1);
  });

  it('should have subcomponent called Amenities', () => {
    expect(shallow(<App />).find(Amenities).length).toEqual(1);
  });

  it('should have subcomponent called Beds', () => {
    expect(shallow(<App />).find(SleepArrangements).length).toEqual(1);
  });

});