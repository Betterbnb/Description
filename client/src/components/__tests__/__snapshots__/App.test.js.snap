// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly with no props 1`] = `
<div
  id="home"
>
  <h1>
    100
  </h1>
  <Header
    amenities={
      Object {
        "basics": Array [
          "smoking allowed",
          "wifi",
          "air conditioning",
          "fireplace",
          "TV",
          "essentials (towels, bedsheets, soap, toilet paper, pillows)",
          "shampoo",
          "parking",
          "iron",
          "pets allowed",
          "heating",
          "desk/workspace",
        ],
        "dining": Array [
          "grill",
          "breakfast",
          "coffee/tea",
          "pizza oven",
        ],
        "facilities": Array [
          "washer",
          "pool",
          "gym",
        ],
        "safety": "safety card",
      }
    }
    bathrooms={15}
    beds={
      Object {
        "bedrooms": Array [
          Array [
            "floor mattress",
            "king",
            "twin",
            "double",
          ],
          "queen",
        ],
        "commonSpace": Array [
          "king",
          "couch",
          "queen",
          "floor mattress",
        ],
      }
    }
    description={
      Object {
        "general": "Adipisicing non sit id dolor et esse nulla qui velit aliqua velit dolor adipisicing anim. Amet consequat excepteur anim esse qui deserunt aute sint aute dolor velit. Labore dolor occaecat aliqua ea veniam ea sit commodo minim ullamco sint veniam. Tempor voluptate sint esse. Cupidatat ad labore laboris irure est commodo. Officia adipisicing velit.",
        "guestAccess": "Mollit esse excepteur dolore nisi minim ea consectetur excepteur Lorem. Amet reprehenderit cupidatat laboris consectetur ipsum anim laboris non laborum. Cillum cillum amet dolore ea excepteur Lorem adipisicing dolor ullamco velit duis. Est anim deserunt eiusmod ex consequat proident consectetur laborum ipsum velit.

Eu Lorem excepteur excepteur reprehenderit tempor tempor quis magna Lorem sit incididunt velit. Nostrud tempor non dolore sint quis nulla culpa labore aliqua ut. Esse magna excepteur non excepteur id eiusmod fugiat velit magna proident. Adipisicing laboris ullamco do amet reprehenderit consectetur magna culpa. Sit magna irure enim consectetur laboris proident officia proident ut irure id.

Et in laboris. Commodo tempor enim deserunt nulla exercitation nostrud anim velit voluptate minim. Anim veniam exercitation. Eu elit ullamco cupidatat. Et Lorem minim nostrud voluptate.",
        "interactionWithGuests": "Occaecat labore voluptate pariatur proident aliquip esse proident ad voluptate eiusmod laborum non. Do ipsum nisi velit veniam non. Amet mollit voluptate esse occaecat exercitation mollit consequat do adipisicing mollit. Anim culpa dolore eiusmod incididunt.

Id adipisicing ad ut sit eiusmod nisi consequat occaecat id. Pariatur eiusmod nulla minim mollit occaecat velit. Nisi pariatur id duis fugiat sint. Eiusmod consequat eiusmod officia minim dolore incididunt in cillum pariatur fugiat. Ea ut dolor incididunt aliqua minim.

Quis enim enim dolore qui consequat duis. Sint incididunt fugiat ex proident ad cillum voluptate sit. Deserunt sint do sunt commodo ad deserunt tempor.",
        "otherThings": "Et nisi irure esse id sit ex eu ullamco excepteur officia amet adipisicing. Aliquip et irure do pariatur qui velit dolore dolor consequat reprehenderit irure. Labore anim officia nostrud excepteur adipisicing qui incididunt.

Quis aliquip culpa dolor minim adipisicing mollit ullamco qui tempor. Aliquip reprehenderit elit minim veniam reprehenderit magna. Cillum et magna sint anim officia proident fugiat Lorem eu. Culpa nostrud culpa cupidatat amet in.

Incididunt sint non sit irure Lorem laborum amet reprehenderit Lorem consequat reprehenderit. Labore sunt mollit amet voluptate commodo sint duis. Laboris aliqua fugiat qui qui sint pariatur ad occaecat qui aliquip voluptate ex occaecat. Amet ullamco ut quis minim eiusmod adipisicing dolore. Est fugiat consectetur irure consectetur duis. Est mollit elit esse voluptate est aliqua minim id dolore sint. Ea labore quis elit.",
        "theSpace": "Duis voluptate tempor velit velit Lorem dolor deserunt consequat ullamco voluptate occaecat nostrud. Sit labore ex id velit. Eu cupidatat do commodo ex enim ad in sint duis consequat aute tempor.

Enim voluptate amet aliquip et est ipsum tempor et enim. Incididunt aliqua est esse qui labore Lorem nisi Lorem aliquip adipisicing nisi. Do in sint do et voluptate qui.

Minim sit cupidatat duis enim consectetur consectetur quis eiusmod labore. Mollit esse cillum quis voluptate veniam. Irure voluptate sunt do ex voluptate est minim ut laboris do. Laboris sunt adipisicing excepteur nisi proident tempor laboris cupidatat laboris excepteur aliqua.",
      }
    }
    guests={9}
    highlights={
      Array [
        Object {
          "description": "This is one of few homes in this area that has this feature.",
          "helpful": 0,
          "notHelpful": 0,
          "title": "Hot tub",
        },
        Object {
          "description": "94% of recent guests gave this home’s value a 5-star rating.",
          "helpful": 0,
          "notHelpful": 0,
          "title": "Great value",
        },
      ]
    }
    homeId={100}
    location="Mandalay, Myanmar"
    miniAd={
      Object {
        "description": "It’s been viewed 500+ times in the past week.",
        "title": "This home is on people’s minds.",
      }
    }
    name="Romantic Cabana"
    owner={
      Object {
        "badge": true,
        "contact": "jpjohnson@gmail.com",
        "image": "https://s3-us-west-1.amazonaws.com/betterbnb-description/person1.jpeg",
        "name": "Michaelangelo",
      }
    }
    propertyType="earth house"
  />
  <PromoBar />
  <Description />
  <Contact />
  <Amenities />
  <SleepArrangements />
</div>
`;
